src/libcore/ops.rs

```diff
@@ -1463,15 +1463,17 @@ pub trait IndexMut<Idx: ?Sized>: Index<Idx> {
 /// # Examples
 ///
 /// ```
-/// fn main() {
-///     assert_eq!((..), std::ops::RangeFull);
+/// assert_eq!((..), std::ops::RangeFull);
 ///
-///     let arr = [0, 1, 2, 3];
-///     assert_eq!(arr[ .. ], [0,1,2,3]);  // RangeFull
-///     assert_eq!(arr[ ..3], [0,1,2  ]);
-///     assert_eq!(arr[1.. ], [  1,2,3]);
-///     assert_eq!(arr[1..3], [  1,2  ]);
-/// }
+/// // for i in .. {
+/// //     println!("This errors because .. has no IntoIterator impl");
```

Comment:
So, usually rather than commenting the code out, we put things in its own example marked with `ignore`. And in general, these feel like three separate examples to me. Could you maybe split them up, and put a little sentence between them explaining it?

So for example:


    /// The `..` syntax is a `RangeFull`:
    ///
    /// ```
    /// assert_eq!((..), std::ops::RangeFull);
    /// ```
    ///
    /// It does not have an `IntoIterator` implementation, so you can't use it in a
    /// `for` loop directly. This won't compile:
    ///
    /// ```ignore
    /// for i in .. {
    ///    // ...
    /// }
    /// ```

Something like that? What do you think?